### ISSUE TITLE
500 Error when metrics.prometheus is enabled

### ISSUE THREAD

**Issue by mavidser** [containous/traefik#3193]
### Do you want to request a *feature* or report a *bug*?
bug

### What did you do?
Enabled `metrics.prometheus`

### What did you expect to see?
Metrics in prometheus, and a working server

### What did you see instead?
Metrics in prometheus, but no working server

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.6.0-rc5
Codename:     tetedemoine
Go version:   go1.10.1
Built:        2018-04-12_09:50:36AM
OS/Arch:      linux/arm
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
debug = true
logLevel = "DEBUG"
[entryPoints]
    [entryPoints.http]
    address = ":80"
[api]
[ping]
[docker]
endpoint = "tcp://192.168.1.8:2375"
[metrics]
[metrics.prometheus]
  entryPoint = "traefik"
  buckets = [0.1,0.3,1.2,5.0]
```

### If applicable, please paste the log output at DEBUG level (`--logLevel=DEBUG` switch)

```
time="2018-04-16T16:54:13Z" level=error msg="Recovered from panic in http handler: runtime error: invalid memory address or nil pointer dereference"
```

**Comment by ldez** [comment]
Could you provide more logs?

**Comment by mavidser** [comment]
Honestly, that's all the logs I'm getting. Full log [without prometheus](https://pastebin.com/uFMdNnXQ), and [with](https://pastebin.com/mkAM1PMr). The only difference is that enabling prometheus starts returning `Internal Server Error`.

Error's not dependent on the running containers as I also tested them without any, and it still displayed the Internal Error instead of _backend not found_ 404.

This is an ARM release, btw. Happening both, inside the container (on `resin/rpi-raspbian:stretch-20180411` image), and natively, on a raspberry pi.

**Comment by mavidser** [comment]
Built it after removing the recover() from `middlewares/recover.go`. Stack trace:

```
http: panic serving 192.168.1.6:61170: runtime error: invalid memory address or nil pointer dereference
goroutine 42 [running]:
net/http.(*conn).serve.func1(0x1305a180)
  /home/sid/go-bin/src/net/http/server.go:1726 +0x9c
panic(0x15e5e88, 0x27fcef8)
  /home/sid/go-bin/src/runtime/panic.go:502 +0x204
sync/atomic.addUint64(0x131564a4, 0x1, 0x0, 0x4, 0x12e23a00)
  /home/sid/go-bin/src/sync/atomic/64bit_arm.go:31 +0x4c
github.com/containous/traefik/middlewares.(*metricsMiddleware).ServeHTTP(0x13156480, 0x1a60c38, 0x131e33c0, 0x12eaa600, 0x12e22e30)
  /home/sid/go/src/github.com/containous/traefik/middlewares/metrics.go:55 +0x15c
github.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x1a56cd8, 0x13156480, 0x12e23b60, 0x1a60c38, 0x131e33c0, 0x12eaa600)
  /home/sid/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x8c
github.com/containous/traefik/vendor/github.com/urfave/negroni.(middleware).ServeHTTP-fm(0x1a60c38, 0x131e33c0, 0x12eaa600)
  /home/sid/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x4c
net/http.HandlerFunc.ServeHTTP(0x12e22db0, 0x1a60c38, 0x131e33c0, 0x12eaa600)
  /home/sid/go-bin/src/net/http/server.go:1947 +0x34
github.com/containous/traefik/middlewares/accesslog.(*LogHandler).ServeHTTP(0x13086140, 0x75515ef8, 0x12e1af10, 0x12eaa580, 0x12e22db0)
  /home/sid/go/src/github.com/containous/traefik/middlewares/accesslog/logger.go:154 +0x8d4
github.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x1a56d08, 0x13086140, 0x12e23b50, 0x75515ef8, 0x12e1af10, 0x12eaa580)
  /home/sid/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x8c
github.com/containous/traefik/vendor/github.com/urfave/negroni.(middleware).ServeHTTP-fm(0x75515ef8, 0x12e1af10, 0x12eaa580)
  /home/sid/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x4c
net/http.HandlerFunc.ServeHTTP(0x12e22da0, 0x75515ef8, 0x12e1af10, 0x12eaa580)
  /home/sid/go-bin/src/net/http/server.go:1947 +0x34
github.com/containous/traefik/middlewares.NegroniRecoverHandler.func1(0x75515ef8, 0x12e1af10, 0x12eaa580, 0x12e22da0)
  /home/sid/go/src/github.com/containous/traefik/middlewares/recover.go:22 +0x60
github.com/containous/traefik/vendor/github.com/urfave/negroni.HandlerFunc.ServeHTTP(0x192f3d0, 0x75515ef8, 0x12e1af10, 0x12eaa580, 0x12e22da0)
  /home/sid/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:24 +0x3c
github.com/containous/traefik/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x1a5a218, 0x192f3d0, 0x12e23b40, 0x75515ef8, 0x12e1af10, 0x12eaa580)
  /home/sid/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:33 +0x8c
github.com/containous/traefik/vendor/github.com/urfave/negroni.(*Negroni).ServeHTTP(0x13087ea0, 0x1a684f8, 0x13200480, 0x12eaa580)
  /home/sid/go/src/github.com/containous/traefik/vendor/github.com/urfave/negroni/negroni.go:81 +0xc4
github.com/containous/traefik/vendor/github.com/containous/mux.(*Router).ServeHTTP(0x13156510, 0x1a684f8, 0x13200480, 0x12eaa580)
  /home/sid/go/src/github.com/containous/traefik/vendor/github.com/containous/mux/mux.go:133 +0xa8
net/http.serverHandler.ServeHTTP(0x12e43780, 0x1a684f8, 0x13200480, 0x12eaa480)
  /home/sid/go-bin/src/net/http/server.go:2694 +0x74
net/http.(*conn).serve(0x1305a180, 0x1a6b618, 0x131e30c0)
  /home/sid/go-bin/src/net/http/server.go:1830 +0x57c
created by net/http.(*Server).Serve
  /home/sid/go-bin/src/net/http/server.go:2795 +0x208
http: panic serving 192.168.1.6:61171: runtime error: invalid memory address or nil pointer dereference
...
similar stack trace with different addresses
...
http: panic serving 192.168.1.6:61184: runtime error: invalid memory address or nil pointer dereference
...
similar stack trace with different addresses
...
```

[Full stack trace](https://pastebin.com/raw/2dBUt0gR)

**Comment by traefiker** [comment]
Closed by #3195.
